src/result.rs

```diff
@@ -70,10 +70,29 @@ impl From<OtherError> for TockError {
     }
 }
 
-pub const SUCCESS: isize = 0;
-pub const FAIL: isize = -1;
-pub const EBUSY: isize = -2;
-pub const EALREADY: isize = -3;
-pub const EINVAL: isize = -6;
-pub const ESIZE: isize = -7;
-pub const ENOMEM: isize = -9;
+#[derive(Copy, Clone, PartialEq, Eq)]
+pub enum ReturnCode {
+    SUCCESS,
+    FAIL,
+    EBUSY,
+    EALREADY,
+    EINVAL,
+    ESIZE,
+    ENOMEM,
+    UNKNOWN(isize),
+}
+
+impl From<isize> for ReturnCode {
+    fn from(return_code: isize) -> Self {
+        match return_code {
+            0 => ReturnCode::SUCCESS,
+            -1 => ReturnCode::FAIL,
+            -2 => ReturnCode::EBUSY,
+            -3 => ReturnCode::EALREADY,
+            -6 => ReturnCode::EINVAL,
+            -7 => ReturnCode::ESIZE,
+            -9 => ReturnCode::ENOMEM,
+            _ => ReturnCode::UNKNOWN(return_code),
```

Comment:
Some values are missing.

Reply:
It's okay that the known variants are in SCREAMING case, s.t. they are consistent with the documentation. The `Unknown` variant, however, should be regular.